reposerver/repository/repository.go

```diff
@@ -1746,12 +1746,12 @@ func getObjsFromYAMLOrJSON(logCtx *log.Entry, manifestPath string, filename stri
 				return status.Errorf(codes.FailedPrecondition, "Failed to unmarshal %q: %v", filename, err)
 			}
 			// Read the whole file to check whether it looks like a manifest.
-			out, err := utfutil.ReadFile(manifestPath, utfutil.UTF8)
-			// Otherwise, let's see if it looks like a resource, if yes, we return error
+			out, _ := utfutil.ReadFile(manifestPath, utfutil.UTF8)
+			// Otherwise, let's see if it looks like a resource, if yes, we log a warning.
 			if bytes.Contains(out, []byte("apiVersion:")) &&
 				bytes.Contains(out, []byte("kind:")) &&
 				bytes.Contains(out, []byte("metadata:")) {
-				return status.Errorf(codes.FailedPrecondition, "Failed to unmarshal %q: %v", filename, err)
+				logCtx.Warnf("Failed to unmarshal %q: file contains 'apiVersion:', 'kind:', and 'metadata:', but is not a valid manifest: %v", filename, err)
```

Comment:
Let's keep the err on line 1749 and log it as part of the warning.

Reply:
@ishitasequeira Thanks for review.
If the file read fails, I think it doesn't make much sense to run the following `if bytes.Contains(out, []byte(...))`. 

I suggest returning an error instead:

```go
out, rerr := utfutil.ReadFile(manifestPath, utfutil.UTF8)
if rerr != nil {
    return status.Errorf(codes.FailedPrecondition, "Failed to read %q: %v", filename, rerr)
    }
```

Let me know what you think.

Reply:
Thanks @CefBoud, that would actually be better and I see that you have already addressed that in your latest commits.